sale_exception/tests/test_sale_exception.py

```diff
@@ -204,3 +204,18 @@ def test_exception_no_free(self):
         so_except_confirm.action_confirm()
         self.assertFalse(sale_order.ignore_exception)
         self.assertTrue(sale_order.state == "draft")
+
+    def test_exception_no_stock_warning(self):
+        partner = self.env.ref("base.res_partner_1")
+        partner.zip = False
+        sale_order = self._create_sale_order(
+            partner=partner, product=self.env.ref("product.product_product_6")
+        )
+        sale_order.action_confirm()
+        self.assertEqual(sale_order.state, "sale")
+        exception = self.env.ref("sale_exception.excep_no_zip")
+        exception.active = True
+        sale_order.action_done()
+        self.assertEqual(sale_order.state, "done")
+        sale_order.action_unlock()
```

Comment:
To make the context clearer.

```suggestion
        # Just reverting the state to 'sale' would trigger an error.
        with self.assertRaises(ValidationError):
            sale_order.write({"state": "sale"})
        # action_unlock(), which also writes the state to 'sale', should be free from validation.
        sale_order.action_unlock()
```

Reply:
@dalonsofl Could you comment the function as suggested?

Maybe:

# Avoid triggering sale exceptions when trying to unlock sale order

Reply:
@dalonsofl

Reply:
Hi @rousseldenis ! I have already added the context, thank you!

Reply:
Oops didn't see. It was above